src/nf/nf_layernorm_submodule.f90

```diff
@@ -0,0 +1,158 @@
+submodule(nf_layernorm_layer) nf_layernorm_layer_submodule
+  implicit none
+contains
+    module function layernorm_layer_cons() &
+    result(res)
+    type(layernorm_layer) :: res
+
+    res % eps = 1e-5
+  end function layernorm_layer_cons
+
+  pure module subroutine forward(self, input)
+    class(layernorm_layer), intent(in out) :: self
+    real, intent(in) :: input(:, :)
+    real, allocatable :: normalized(:, :)
+    integer :: i
+
+    allocate(normalized, mold=self % mu)
```

Comment:
Since the shape of `mu` is determined at `init`, so is the shape of `normalized`.  It's inefficient to allocate+deallocate an array in every `forward` call. Let's allocate `normalized` in `init` as well.